lib/utils/date_utils.ts

```diff
@@ -71,11 +71,67 @@ function createEdgeDate(edge : ('start_of'|'end_of'), unit : string, timezone :
     return new Date(date.epochMilliseconds);
 }
 
-function createDatePiece(year : number|null, month : number|null, day : number|null, time : AbsoluteTime|null, timezone : string) : Date {
+function getTimeMs(hour : number, minute : number, second : number) {
+    const hourMs = hour * 1000 * 60 * 60;
+    const minMs = minute * 1000 * 60;
+    const secMs = second * 1000;
+    return hourMs + minMs + secMs;
+}
+
+function createDatePiece(year : number|null, month : number|null, day : number|null, time : AbsoluteTime|null, timezone : string, future ?: boolean) : Date {
+    let date = Temporal.Now.zonedDateTimeISO(timezone);
+    //Get the next instance of the supplied values
+    //Use the last blank value to get the next instance
+    if (future) {
+        if (day === null) {
+            if (time !== null) {
+                const curTime = getTimeMs(date.hour, date.minute, date.second);
+                const timeMs = getTimeMs(time.hour, time.minute, time.second);
+                //if the desired time is in the past for this day, get tomorrow
+                if (curTime > timeMs) {
+                    date.withPlainDate({
+                        year: date.year,
+                        month: date.month,
+                        day: date.day + 1
+                    });
```

Comment:
This code is not correct to compute tomorrow, you need to use `.add({ days: 1 })` because the month/year might change too.